Sources/Extensions/UIBezierPathExtension.swift

```diff
@@ -162,10 +162,148 @@ extension UIBezierPath {
 
     close()
   }
+
+  /**
+   Create a Bezier path for a heart shape.
+
+   - Parameter bounds: The bounds of shape.
+   */
+  convenience init(heartIn bounds: CGRect)  {
+    self.init()
+    var x: CGFloat = bounds.origin.x
+    var y: CGFloat = bounds.origin.y
+
+    // square bounds
+    let width = ceil(min(bounds.size.width, bounds.size.height))
+    let height = width
+    x = x + (bounds.size.width - width)/2
+    y = y + (bounds.size.height - height)/2
+
+    let lowerPoint = CGPoint(x: x + width / 2, y: (y + height ))
+    move(to: lowerPoint)
+
+    addCurve(to: CGPoint(x: x, y: (y + (height/4))),
+             controlPoint1:CGPoint(x: (x + (width/2)), y: (y + (height*3/4))),
+             controlPoint2: CGPoint(x: x, y: (y + (height/2))))
+
+    addArc(withCenter: CGPoint(x: (x + (width/4)), y: (y + (height/4))),
+           radius: (width/4),
+           startAngle: .pi,
+           endAngle: 0,
+           clockwise: true)
+
+    addArc(withCenter: CGPoint(x: (x + (width * 3/4)), y: (y + (height/4))),
+           radius: (width/4),
+           startAngle: .pi,
+           endAngle: 0,
+           clockwise: true)
+
+    addCurve(to: lowerPoint,
+             controlPoint1: CGPoint(x: (x + width), y: (y + (height/2))),
+                  controlPoint2: CGPoint(x: (x + (width/2)), y: (y + (height*3/4))) )
+  }
+
+  /**
+   Create a Bezier path for a ring shape.
+   
+   - Parameter bounds: The bounds of shape.
+   - Parameter radius: The radius of the shape.
+   */
+  convenience init(ringIn bounds: CGRect, radius: CGFloat) {
+    let center = CGPoint(x: bounds.midX, y: bounds.midY)
+    let diameter = ceil(min(bounds.width, bounds.height))
+    let innerRadius = max(1, diameter / 2 - radius)
+    let outerRadius = diameter / 2
+
+    self.init()
+    addArc(withCenter: .zero, radius: innerRadius, startAngle: 0, endAngle: .pi * 2, clockwise: true)
+
+    move(to: CGPoint(x: outerRadius, y: 0))
+
+    addArc(withCenter: .zero, radius: outerRadius, startAngle: 0, endAngle: .pi * 2, clockwise: true)
+
+    apply(CGAffineTransform(translationX: center.x, y: center.y))
+    usesEvenOddFillRule = true
+  }
+
+  /**
+   Create a Bezier path for a gear shape.
+
+   - Parameter bounds: The bounds of shape.
+   - Parameter radius: The radius of the shape.
+   - Parameter cogs: The number of cogs (min: 2)
+   */
+  convenience init(gearIn bounds: CGRect, radius: CGFloat, cogs: Int) {
+    let center = CGPoint(x: bounds.midX, y: bounds.midY)
+    let diameter = ceil(min(bounds.width, bounds.height))
+    let innerRadius = max(1, diameter / 2 - radius)
+    let outerRadius = diameter / 2
```

Comment:
We may be able to encapsulate these four lines into a method because they are same as line 231.

Reply:
You want something like this ?
```swift
let (center, diameter, innerRadius, outerRadius) = rect.ringParameters(radius)
```

I can make CGRect extension
like 
```swift
var center: CGPoint { // I do not know why this do not exist already in CoreGraphic
  return CGPoint(x: self.midX, y: self.midY)
}
var diameter: CGFloat {
  return ceil(min(self.width, self.height))
}
```

Reply:
I prefer to have extensions. They look better to me.